zerver/lib/upload.py

```diff
@@ -190,10 +207,22 @@ def upload_avatar_image_local(user_file, user_profile, email):
 if settings.LOCAL_UPLOADS_DIR is not None:
     upload_message_image = upload_message_image_local
     upload_avatar_image  = upload_avatar_image_local
+
 else:
     upload_message_image = upload_message_image_s3
     upload_avatar_image  = upload_avatar_image_s3
 
+
+def claim_attachment(path_id, message):
+    #update attachment entry in DB
+    try:
+        Attachment.objects.filter(path_id=path_id).update(message=message)
+        return True
+    except Attachment.DoesNotExist:
+        raise JsonableError("The upload was not successful. Please reupload the file again in a new message.")
```

Comment:
Can you add a test that this exception is what you get if the path ID isn't present?  Seems like should be a fairly simple test to write -- just try sending a message with a URL of the right form without having first uploaded a file.